data/core/units/monsters/Fungoid_Body.cfg

```diff
@@ -0,0 +1,275 @@
+            amount=$damage_inflicted
+            animate=yes
+        [/heal_unit]
+    [/event]
+    [event]
+        name=defender hits
+        first_time_only=no
+        [filter_second]
+            type=Fungoid Spore,Fungoid Stalk
+        [/filter_second]
+        [heal_unit]
+            [filter]
+                type=Fungoid Body
+            [/filter]
+            amount=$damage_inflicted
+            animate=yes
+        [/heal_unit]
+    [/event]
+
+    # Every turn there is a chance a fruiting body will produce a new spore unit.
+    [event]
+        name=turn end
+        first_time_only=no
+        [store_unit]
+            [filter]
+                type=Fungoid Body
+            [/filter]
+            variable=bodies
+        [/store_unit]
+        [foreach]
+            array=bodies
+            variable=body
+            [do]
+                [store_unit]
+                    [filter]
+                        type=Fungoid Stalk
+                        side=$body.side
+                    [/filter]
+                    variable=stalks
+                [/store_unit]
+                [store_unit]
+                    [filter]
+                        type=Fungoid Spore
+                        side=$body.side
+                    [/filter]
+                    variable=spores
+                [/store_unit]
+                [if]
+                    # Sporulation rate increases with the number of stalks but decreases with the number of existing spores.
+                    [lua]
+                        code=<< local t = ...; return ( mathx.random() < ((t.stalks + 1) * 3 / (t.spores + 1) / wesnoth.unit_types["Fungoid Spore"].cost) ) >>
+                        [args]
+                            stalks=$stalks.length
+                            spores=$spores.length
+                        [/args]
+                    [/lua]
+                    [then]
+                        [unit]
+                            side=$body.side
+                            type=Fungoid Spore
+                            placement=map
+                            x,y=$body.x,$body.y
+                            passable=yes
+                            animate=yes
+                        [/unit]
+                    [/then]
+                [/if]
+            [/do]
+        [/foreach]
+    [/event]
+
+    [attack]
+        name=glow
+        description= _ "glow"
+        icon=attacks/energyray.png
+        type=cold
+        range=melee
+        damage=10
+        number=3
+        [specials]
+            {WEAPON_SPECIAL_STUN}
+        [/specials]
+    [/attack]
+    [attack]
+        name=glow
+        description= _"glow"
+        icon=attacks/energyray.png
+        type=cold
+        range=ranged
+        damage=10
+        number=3
+        [specials]
+            {WEAPON_SPECIAL_STUN}
+        [/specials]
+    [/attack]
+    [attack_anim]
+        # The following attack filter deliberately matches both ranged and melee.
```

Comment:
Using the same animation for two different attacks seems bad… as does having two identical attacks that differ only in range.

Reply:
It's fine for the melee attack to not lunge, but I really don't think it should be the _same_ animation as the ranged attack. Some example possibilities:

* Ranged is a missile (thorn maybe?), melee is a tentacle attack
* Ranged is a really long tentacle, melee is a shorter tentacle
* Ranged is a tentacle, melee just sort of headbutts without moving at the base

Reply:
In terms of visual design, the headbutt could potentially work for the Stalk unit but maybe not the Body unit, while a tentacle attack could work for both (and it might also have the `slows` weapon special, which would be interestingly unique on a melee attack).

Have you seen the existing approach in game, though? Like while playing the final scenario of UtBS? It may just be me, but I find it suits these creatures, that they solemnly stand and cast this one attack from any distance. It kind of gives them an air of otherworldly power and indifference (which persists even when they are attacked up close).